File: Leetcode/FloodFill.cpp
class Solution {
public:
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        queue <pair <int, int>> q;
        int vis[52][52], nx, ny;
        pair <int, int> u, v;
        int dx[] = {1, 0, 0, -1};
        int dy[] = {0, 1, -1, 0};
        int n = image.size(), m = image[0].size();
        int cl = image[sr][sc];
        image[sr][sc] = color;
    
        q.push({sr, sc});
        
        for(int i = 0; i < 52; i++)
            for(int j = 0; j < 52; j++)
                vis[i][j] = 0;
        
        vis[sr][sc] = 1;
        
        while(!q.empty()){
            u = q.front();
            
            for(int i = 0; i < 4; i++){
                nx = dx[i] + u.first;
                ny = dy[i] + u.second;
                
                if(nx >= 0 && nx < n && ny >= 0 && ny < m && !vis[nx][ny] && image[nx][ny] == cl){
                    vis[nx][ny] = 1;
                    q.push({nx, ny});
                    image[nx][ny] = color;
                }
            }
            
            q.pop();
        }
        return image;
    }
};

File: Leetcode/FloodFill.py
class Solution(object):
    def floodFill(self, image, sr, sc, color):
        mat = image
        n = len(image)
        m = len(image[0])
        vis = [[0 for _ in range(m)] for _ in range(n)]
        vis[sr][sc] = 1
        dx = [0, 1, 0, -1]
        dy = [1, 0, -1, 0]
        q = [(sr, sc)]
        
        while len(q) != 0:
            x, y = q.pop(0)
            
            for i in range(4):
                nx = dx[i] + x
                ny = dy[i] + y
                
                if nx >= 0 and nx < n and ny >= 0 and ny < m and vis[nx][ny] == 0:
                    vis[nx][ny] = 1
                    if mat[nx][ny] == mat[sr][sc]:
                        mat[nx][ny] = color
                        q.append((nx, ny))
        mat[sr][sc] = color
        return image
                        